### PR TITLE
feat(protocol): add impact data

### PR DESCRIPTION
**Pull request by maxd-nordic**
This adds support for motion impact data following the implementation
in https://github.com/nrfconnect/sdk-nrf/pull/7946.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

### CODE REVIEW

File: docs/cloud-protocol/messages.schema.json [hunk 1]
```diff
@@ -23,6 +23,25 @@
...
       "required": ["v", "ts"]
+    },
+    "impact": {
+      "description": "The magnitude and time of a motion impact event",
+      "type": "object",
+      "properties": {
+        "v": {
+          "description": "Magnitude of the impact in g",
+          "type": "number",
+          "minimum": 0.5,
```
Comment (coderbyheart) [inline docs/cloud-protocol/messages.schema.json]: **question:** Why 0.5 g minimum?

**Review by coderbyheart — COMMENTED** [review]
I understand that [the new firmware feature](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R89-R94) has two values / events?
1. [Impact](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R94) (which is expressed in this PR)
2. [Peak magnitude](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R90)

Can we add the peak magnitude as well?

**Comment by maxd-nordic** [comment]
> I understand that [the new firmware feature](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R89-R94) has two values / events?
> 
>     1. [Impact](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R94) (which is expressed in this PR)
> 
>     2. [Peak magnitude](https://github.com/nrfconnect/sdk-nrf/pull/7946/files#diff-86cef2135b9163a292238fa403527140e68b48ecd952e2e4e18e26c3edd0f237R90)
> 
> 
> Can we add the peak magnitude as well?

Both are coupled. Impact events always carry a peak magnitude value to describe how strong the impact was. This is the value we see here.

File: docs/cloud-protocol/messages.schema.json (hunk 1 above)
Comment (maxd-nordic) [reply docs/cloud-protocol/messages.schema.json]: I set it on 0.0 now, since it is an absolute value.

**Review by coderbyheart — APPROVED** [review]
Checks must pass, otherwise OK.

**Pull request closed without merging**